modules/openapi-generator/src/main/resources/python/python-experimental/api_client.mustache

```diff
@@ -542,3 +561,180 @@ class ApiClient(object):
+                request_body = body.encode()
+                body_digest, digest_prefix = self.get_message_digest(request_body)
+                b64_body_digest = b64encode(body_digest.digest())
+                value = digest_prefix + b64_body_digest.decode('ascii')
+                signed_header_dict['Digest'] = '{0}{1}'.format(digest_prefix, b64_body_digest.decode('ascii'))
+            elif hdr_key == 'host':
+                value = target_host
+                signed_header_dict['Host'] = '{0}'.format(target_host)
+            else:
+                value = headers[hdr_key]
+            signed_headers[hdr_key] = value
+
+        if len(self.configuration.signed_headers) == 0:
+            signed_headers['(created)'] = created
+
+        string_to_sign = self.get_str_to_sign(signed_headers)
+
+        digest, digest_prefix = self.get_message_digest(string_to_sign.encode())
+        b64_signed_msg = self.sign_digest(digest)
+
+        auth_header = self.get_authorization_header(signed_headers, b64_signed_msg, created)
+        signed_header_dict['Authorization'] = '{0}'.format(auth_header)
+
+        return signed_header_dict
+
+    def get_message_digest(self, data):
+        """
+        Calculates and returns a cryptographic digest of a specified HTTP request.
+
+        :param data: The message to be hashed with a cryptographic hash.
+        :return: The message digest encoded as a byte string. 
+        """
+        if self.configuration.signing_scheme in ["rsa-sha512", "hs2019"]:
+            digest = SHA512.new()
+            prefix = "SHA-512="
+        elif self.configuration.signing_scheme in ["rsa-sha256"]:
+            digest = SHA256.new()
+            prefix = "SHA-256="
+        else:
+            raise Exception(
+                "Unsupported signing algorithm: {0}".format(self.configuration.signing_scheme))
+        digest.update(data)
+        return digest, prefix
+
+    def sign_digest(self, digest):
+        """
+        Signs a message digest with a private key specified in the configuration.
+
+        :param digest: digest of the HTTP message.
+        :return: the HTTP message signature encoded as a byte string.
+        """
+        self.configuration.load_private_key()
+        privkey = self.configuration.private_key
+        if isinstance(privkey, RSA.RsaKey):
+            if self.configuration.signing_algorithm == 'PSS':
+                # RSASSA-PSS in Section 8.1 of RFC8017.
+                signature = pss.new(privkey).sign(digest)
+            elif self.configuration.signing_algorithm == 'PKCS1v15':
+                # RSASSA-PKCS1-v1_5 in Section 8.2 of RFC8017.
+                signature = PKCS1_v1_5.new(privkey).sign(digest)
+            else:
+                raise Exception("Unsupported signature algorithm: {0}".format(self.configuration.signing_algorithm))
+        elif isinstance(privkey, ECC.EccKey):
+            if self.configuration.signing_algorithm in ['fips-186-3', 'deterministic-rfc6979']:
```

Comment:
Can you turn these into a constant at the top of this module?
Something like:
DSS_SIGNING_ALGORITHMS  {'fips-186-3', 'deterministic-rfc6979'}
Set inclusion should be quicker than list inclusion and these values are unchanging across multiple function calls

Reply:
doing.